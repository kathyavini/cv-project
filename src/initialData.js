const data = {
  information: {
    name: 'Simran Kaur',
    title: 'Fullstack developer',
    location: 'Ithica, NY',
    github: 'github.com/simran_kaur',
    email: 'skaur@email.com',
    phone: '555-555-555',
  },
  skills: [
    {
      title: 'Languages',
      body: ['JavaScript/Node.js', 'CSS3', 'HTML5'],
    },
    {
      title: 'Libraries',
      body: ['React', 'D3', 'Express'],
    },
    {
      title: 'Other Tools',
      body: ['Git', 'Jest', 'Cypress'],
    },
  ],
  experience: [
    {
      title: 'Cornell Lab of Ornithology - eBird Project',
      subtitle: 'Applications Programmer',
      dates: '2020-present',
      body: [
        'Contributed to developing and maintaining the eBird web application database and API','Many other important tasks'
      ],
    },
    {
      title: 'Cornell Lab of Ornithology Macaulay Library',
      subtitle: 'Database Assistant',
      dates: '2018-2020',
      body: [
        'Assisted with storage, categorization, and retrieval of bird sounds',
        'Member of engineering team bringing SoundID to the Merlin Bird ID application',
      ],
    },
  ],
  projects: [
    {
      title: 'Mozilla - MDN Web Docs (Open Source Contributor)',
      subtitle: 'Contributed sample code for lessons on mobile app development and React',
    },
  ],
  education: [
    {
      title: 'University of British Columbia',
      subtitle: 'Bachelor of Science Honour Degree in Ecology',
    },
  ],
};

export default data;