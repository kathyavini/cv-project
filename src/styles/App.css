* {
  /* border: 1px solid lightcoral; */
}

.container {
  width: 100vw;
  max-width: 1000px;
  min-height: 100vh;
  background-color: var(--black);
  overflow-x: hidden;
}

.resume {
  width: 100%;
  max-width: 800px;
  height: 100%;
  padding: 3vw;
  padding-right: 5vw;
  margin-inline: auto;
}

.information, .skills {
  position: relative;
  margin: 0;
  padding: 0;
}

.reset {
  margin-top: 2rem;
  margin-bottom: 1rem;
  padding-block: 15px;
  padding-inline: 30px;
  width: max-content;
  border: 2px solid var(--accent5);
  color: var(--accent5);
  font-weight: bold;
  border-radius: 5px;
  font-size: 1.1rem;
}

.item-title {
  color: var(--accent3)
}

.item-subtitle {
  color: var(--accent)
}

.projects-main {
  color: var(--accent2)
}

.education-main {
  color: var(--accent6)
}

/* Interactivity */
.edit-button {
  position: absolute;
  top: 0px;
  right: 20px;
  font-size: 36px;
  color: var(--white);
}

.gradient { /* For icons */
  background-image: var(--icon-gradient);
  background-color: var(--icon-fallback);
  color: transparent;
  background-clip: text;
  -webkit-background-clip: text;
}


form {
  outline: 2px solid var(--accent7);
  padding: 10px;
  border-radius: 3px;
}

button  {
  width: 100px;
  font-weight: bold;
  font-size: 1.4rem;
  padding-inline: 12px;
  padding-block: 6px;
  border-radius: 5px;
  min-width: max-content;
  color: var(--black);
  background-image: var(--icon-gradient);
  background-color: var(--icon-fallback);
  border: none;
  outline: none;
  transition: filter 250ms;
  display: block;
  margin-left: auto;
  margin-top: 15px;
  margin-bottom: 5px;
  font-family: var(--secondary-font);
}

button:hover {
  outline: 2px solid var(--light4);
}

button:active {
  filter: brightness(2);
}

input {
  width: 85%;
  background-color: transparent;
  color: var(--text-color);
  font-family: var(--main-font);
  border: none;
  outline: none;
  transition: outline 500ms;
  padding: 10px 15px;
  margin-inline: 5px;
  border-radius: 3px;
  font-size: 1.3rem;
}

input:focus {
  outline: 1px solid var(--accent);
  transition: outline 500ms;
}

input + span {
  color: var(--accent1);
}

input + span:active {
  filter: brightness(1.3);
}

.add-item {
  margin-right: auto;
  margin-left: 10%;
  background-color: transparent;
  background-image: none;
  color: var(--accent6);
  border: 2px solid var(--accent6);
  display: flex;
  align-items: center;
  padding-left: 5px;
  font-size: 1.2rem;
}

.add-item span {
  margin-right: 5px;
}

.delete-section {
  display: flex;
  align-items: center;
  font-size: 1.4rem;
  background-image: none;
  background-color: var(--accent4);
}

.delete-section span {
  margin-right: 5px;
}

.add-section {
  margin-right: auto;
  margin-left: 0;
  background-color: transparent;
  background-image: none;
  color: var(--accent1);
  border: 2px solid var(--accent1);
  display: flex;
  align-items: center;
  padding-left: 5px;
  font-size: 1.4rem;
}

.add-section span {
  margin-right: 5px;
}

.component-buttons {
  display: flex;
}

/* Information section */
.name {
  color: var(--accent7)
}

.title {
  color: var(--accent4)
}

.section-title {
  color: var(--accent1)
} 

.contact{
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}

.contact {
  margin-right: auto;
}

/* Skills section */
.skill-group {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
}

.skill-title {
  margin-right: auto;
}

.skill-list {
  padding-left: 5px;
  display: flex;
  flex-flow: row wrap;
  column-gap: 10px;
  grid-column: 3 / 4;
}

@media (hover:none) {
  /* .edit-button {
    display: none;
  } */
}