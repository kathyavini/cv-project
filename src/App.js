import React, { useState } from 'react';
import './styles/App.css';
import initialData from './initialData';
import Information from './components/Information';


function Skills({ skills }) {
  const [editing, setEditing] = useState(false);
  const [activeMode, setActiveMode] = useState(false);

  function handleMouseEnter(event) {
    setActiveMode(true);
  }

  function handleMouseLeave(event) {
    setActiveMode(false);
  }

  return (
    <div
      className={`skills ${activeMode && 'active'}`}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <h2 className="section-title">SKILLS</h2>
      {skills.map((section, index) => (
        <div className="skill-group" key={index}>
          <h3 className="skill-title">{section.title}</h3>
          <div className="skill-list">
            {section.body.map((item, itemIndex) => (
              <React.Fragment key={itemIndex}>
                <p className="skill-item">{item}</p>
                <p>{itemIndex !== section.body.length - 1 && 'Â·'}</p>
              </React.Fragment>
            ))}
          </div>
        </div>
      ))}
    </div>
  );
}

function Experience({ experience }) {
  return (
    <div className="experience">
      <h2 className="section-title">EXPERIENCE</h2>
      {experience.map((job) => {
        return (
          <div className="experience-group" key={job.title}>
            <h3>{job.title}</h3>
            <h4>
              {job.subtitle} | {job.dates}
            </h4>
            <ul key={`points${job.title}`}>
              {job.body.map((task, index) => (
                <li key={index}>{task}</li>
              ))}
            </ul>
          </div>
        );
      })}
    </div>
  );
}

function Projects({ projects }) {
  return (
    <div className="projects">
      <h2 className="section-title">PROJECTS</h2>
      {projects.map((project) => (
        <div className="project-group" key={project.title}>
          <h3>{project.title}</h3>
          <h4>{project.subtitle}</h4>
        </div>
      ))}
    </div>
  );
}

function Education({ education }) {
  return (
    <div className="education">
      <h2 className="section-title">EDUCATION</h2>
      {education.map((school) => (
        <div className="education-group" key={school.title}>
          <h3>{school.title}</h3>
          <h4>{school.subtitle}</h4>
        </div>
      ))}
    </div>
  );
}

function App() {
  const [data, setData] = useState(
    () => JSON.parse(localStorage.getItem('savedData')) || initialData
  );

  React.useEffect(() => {
    localStorage.setItem('savedData', JSON.stringify(data));
  }, [data]);

  // // This is a little local storage functionality test
  // function dummyChange() {
  //   const newInformation = { ...data.information };
  //   newInformation.name = 'Clicked!';
  //   setData({ ...data, information: newInformation });
  // }

  function resetData() {
    setData(initialData);
  }

  return (
    <div className="container">
      <div className="resume">        
      <h3 className="reset" onClick={resetData}>Reset Data</h3>
        <Information information={data.information} setData={setData} data={data}/>
        <Skills skills={data.skills} />
        <Experience experience={data.experience} />
        <Projects projects={data.projects} />
        <Education education={data.education} />

      </div>
    </div>
  );
}

export default App;
