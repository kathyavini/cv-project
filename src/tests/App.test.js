import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import App from '../App';
import initialData from '../initialData';

// Should I mock local storage for the purpose of the snapshot?
test('Initial page load snapshot', () => {
  const { container, getByText } = render(<App />, { legacyRoot: true });
  expect(container).toMatchSnapshot();
});

// Some dummy tests below to remind myself how userEvent works
test('Hover displays edit button', () => {
  render(<App />);

  const input = screen.getByRole('textbox');
  const submitBtn = screen.getByRole('button');

  userEvent.type(input, 'Test task');
  userEvent.click(submitBtn);

  const firstItem = screen.getByRole('listitem');
  expect(firstItem.textContent).toMatch(/test task/i);
});

// Test the reset buttons
test.skip('Clicking on delete icon deletes task', () => {
  render(<App />);
  const input = screen.getByRole('textbox');
  const submitBtn = screen.getByRole('button');

  userEvent.type(input, 'First task');
  userEvent.click(submitBtn);
  userEvent.type(input, 'Second task');
  userEvent.click(submitBtn);
  userEvent.type(input, 'Third task');
  userEvent.click(submitBtn);

  const deleteIcons = screen.getAllByLabelText('delete');

  userEvent.click(deleteIcons[1]);

  const secondTask = screen.queryByText('Second task');

  expect(secondTask).not.toBeInTheDocument();
});
