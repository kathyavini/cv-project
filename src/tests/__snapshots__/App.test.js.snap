// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial page load snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="resume"
    >
      <h3
        class="reset"
      >
        Reset Data
      </h3>
      <div
        class="information"
      >
        <h1
          class="name"
        >
          Simran Kaur
        </h1>
        <h2
          class="title"
        >
          Fullstack developer
        </h2>
        <h3
          class="contact"
        >
          Ithica, NY
           | 
          github.com/simran_kaur
           | 
          skaur@email.com
           | 
          555-555-555
        </h3>
      </div>
      <div
        class="skills"
      >
        <h2
          class="section-title"
        >
          SKILLS
        </h2>
        <div
          class="skill-group"
        >
          <h3
            class="skill-title"
          >
            Languages
          </h3>
          <div
            class="skill-list"
          >
            <p
              class="skill-item"
            >
              JavaScript/Node.js
            </p>
            <p>
              ·
            </p>
            <p
              class="skill-item"
            >
              CSS3
            </p>
            <p>
              ·
            </p>
            <p
              class="skill-item"
            >
              HTML5
            </p>
            <p />
          </div>
        </div>
        <div
          class="skill-group"
        >
          <h3
            class="skill-title"
          >
            Libraries
          </h3>
          <div
            class="skill-list"
          >
            <p
              class="skill-item"
            >
              React
            </p>
            <p>
              ·
            </p>
            <p
              class="skill-item"
            >
              D3
            </p>
            <p>
              ·
            </p>
            <p
              class="skill-item"
            >
              Express
            </p>
            <p />
          </div>
        </div>
        <div
          class="skill-group"
        >
          <h3
            class="skill-title"
          >
            Other Tools
          </h3>
          <div
            class="skill-list"
          >
            <p
              class="skill-item"
            >
              Git
            </p>
            <p>
              ·
            </p>
            <p
              class="skill-item"
            >
              Jest
            </p>
            <p>
              ·
            </p>
            <p
              class="skill-item"
            >
              Cypress
            </p>
            <p />
          </div>
        </div>
      </div>
      <div
        class="experience"
      >
        <h2
          class="section-title"
        >
          EXPERIENCE
        </h2>
        <div
          class="experience-group"
        >
          <h3>
            Cornell Lab of Ornithology - eBird Project
          </h3>
          <h4>
            Applications Programmer
             | 
            2020-present
          </h4>
          <ul>
            <li>
              Contributed to developing and maintaining the eBird web application database and API
            </li>
            <li>
              Many other important tasks
            </li>
          </ul>
        </div>
        <div
          class="experience-group"
        >
          <h3>
            Cornell Lab of Ornithology Macaulay Library
          </h3>
          <h4>
            Database Assistant
             | 
            2018-2020
          </h4>
          <ul>
            <li>
              Assisted with storage, categorization, and retrieval of bird sounds
            </li>
            <li>
              Member of engineering team bringing SoundID to the Merlin Bird ID application
            </li>
          </ul>
        </div>
      </div>
      <div
        class="projects"
      >
        <h2
          class="section-title"
        >
          PROJECTS
        </h2>
        <div
          class="project-group"
        >
          <h3>
            Mozilla - MDN Web Docs (Open Source Contributor)
          </h3>
          <h4>
            Contributed sample code for lessons on mobile app development and React
          </h4>
        </div>
      </div>
      <div
        class="education"
      >
        <h2
          class="section-title"
        >
          EDUCATION
        </h2>
        <div
          class="education-group"
        >
          <h3>
            University of British Columbia
          </h3>
          <h4>
            Bachelor of Science Honour Degree in Ecology
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
